src/main/java/hudson/remoting/Channel.java

```diff
@@ -194,6 +190,14 @@
      * and therefore should be OK.
      */
     private final WeakHashMap<PipeWindow.Key, WeakReference<PipeWindow>> pipeWindows = new WeakHashMap<PipeWindow.Key, WeakReference<PipeWindow>>();
+    /**
+     * There are cases where complex object cycles can cause a closed channel to fail to be garbage collected,
+     * these typically arrise when an {@link #export(Class, Object)} is {@link #setProperty(Object, Object)}
+     * (a supported and intended use case), the {@link Ref} allows us to break the object cycle on channel
+     * termination and simplify the circles into chains which can then be collected easily by the garbage collector.
+     * @since FIXME after merge
```

Comment:
No `@since` needed on non-API members. Anyone doing code archaeology should just `git-blame`.